### PR TITLE
Default retain values for SDK schedule policy

### PR DESCRIPTION
**Pull request by ipochi**
Signed-off-by: Imran Pochi <imran@infracloud.io>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by lpabon** [comment]
Once this is done, this needs to be backported to release-0.9 branch

File: api/api.proto [hunk 1]
```diff
@@ -1543,6 +1543,23 @@ message SdkSchedulePolicyIntervalPeriodic{
 
 // Defines a schedule policy interval
 message SdkSchedulePolicyInterval {
+
+  enum DefaultDailyScheduleRetain {
```
Comment (lpabon) [inline api/api.proto]: We need documentation in all of these. For the ones with UNKNOWN you do not need to prefix them. just call them `UNKOWN` and add a comment `Do not use`.

File: api/server/sdk/schedule_policy_test.go [hunk 2]
```diff
@@ -143,6 +143,115 @@ func TestSdkSchedulePolicyCreateSuccess(t *testing.T) {
...
+
+			// Verify data is correct
+			policies, err := retainInternalSpecYamlByteToSdkSched([]byte(schedule))
+			assert.NoError(t, err)
+			assert.Len(t, policies, len(req.GetSchedulePolicy().GetSchedules()))
+
+			// Check Daily data
+			actualDaily := policies[0]
+
+			fmt.Println(actualDaily.GetRetain())
```
Comment (lpabon) [inline api/server/sdk/schedule_policy_test.go]: remove fmt.Println debugging functions

File: api/server/sdk/schedule_policy_test.go [hunk 3]
```diff
@@ -143,6 +143,115 @@ func TestSdkSchedulePolicyCreateSuccess(t *testing.T) {
...
+			expectedDaily := req.GetSchedulePolicy().GetSchedules()[0]
+			fmt.Println(expectedDaily.GetRetain())
+
+			assert.Equal(t, expectedDaily.GetRetain(), actualDaily.GetRetain())
```
Comment (lpabon) [inline api/server/sdk/schedule_policy_test.go]: So I guess when the SDK calls schedpolicy it does it with retain of 0 still, right?

File: api/api.proto (hunk 1 above)
Comment (ipochi) [reply api/api.proto]: can only add UNKOWN once , gives error when i add again.
Hence i had to prefix

**Comment by lpabon** [comment]
completely forgot to mention that you will need to update the SDKVersion and SDK_CHANGELOG.md accordingly

**Comment by ggriffiths** [comment]
Closing manually as we're closing old PRs before Jan 2020. Feel free to update and re-open if this is still relevant.

**Pull request closed without merging**